src/tournament/controllers/tournament.controller.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Inject,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { Tournament } from '../models/tournament.model';
+import { ITournamentService } from '../interfaces/tournament.service.interface';
+
+@Controller('tournament')
+export class TournamentController {
+  constructor(
+    @Inject('ITournamentService') private serviceInterface: ITournamentService,
```

Comment:
Este simbolo usado no `@Inject` não é necessário para este caso.

```suggestion
    @Inject() private service: ITournamentService,
```